src/core/shape_inference/include/generate_proposals_shape_inference.hpp

```diff
@@ -22,106 +22,120 @@ std::vector<TRShape> shape_infer(const GenerateProposals* op, const std::vector<
-                                  im_info_shape[0]);
+            NODE_SHAPE_INFER_CHECK(op,
+                                   input_shapes,
+                                   deltas_shape[0].compatible(im_info_shape[0]),
+                                   "Batch for inputs 'im_info' and 'input_deltas' should be "
+                                   "equal. Got: ",
+                                   deltas_shape[0],
+                                   im_info_shape[0]);
         }
     }
 
     if (scores_shape_rank.is_static() && anchors_shape_rank.is_static()) {
-        NODE_VALIDATION_CHECK(op,
-                              anchors_shape[0].compatible(scores_shape[2]),
-                              "Heights for inputs 'input_anchors' and 'input_scores' should be "
-                              "equal. Got: ",
-                              anchors_shape[0],
-                              scores_shape[2]);
-
-        NODE_VALIDATION_CHECK(op,
-                              anchors_shape[1].compatible(scores_shape[3]),
-                              "Width for inputs 'input_anchors' and 'input_scores' should be "
-                              "equal. Got: ",
-                              anchors_shape[1],
-                              scores_shape[3]);
-
-        NODE_VALIDATION_CHECK(op,
-                              anchors_shape[2].compatible(scores_shape[1]),
-                              "Anchor number for inputs 'input_anchors' and 'input_scores' should be "
-                              "equal. Got: ",
-                              anchors_shape[2],
-                              scores_shape[1]);
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               anchors_shape[0].compatible(scores_shape[2]),
+                               "Heights for inputs 'input_anchors' and 'input_scores' should be "
+                               "equal. Got: ",
+                               anchors_shape[0],
+                               scores_shape[2]);
+
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               anchors_shape[1].compatible(scores_shape[3]),
+                               "Width for inputs 'input_anchors' and 'input_scores' should be "
+                               "equal. Got: ",
+                               anchors_shape[1],
+                               scores_shape[3]);
+
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               anchors_shape[2].compatible(scores_shape[1]),
+                               "Anchor number for inputs 'input_anchors' and 'input_scores' should be "
+                               "equal. Got: ",
+                               anchors_shape[2],
+                               scores_shape[1]);
     }
 
     auto num_rois = Dimension(0, (num_batches * op->get_attrs().post_nms_count).get_max_length());
```

Comment:
for template usage this conversion should be applied for PartialShape only.

Reply:
The whole shape infer for GenerateProposals is compatible with PartialShape class only, as it always produces dynamic shape.